### PR TITLE
Fetch resources automatically, experimental field support

### PR DESCRIPTION
**Pull request by mattgrill**
- Fetch available resources automatically
- Add experimental support for setting field values.
- Updated documentation.
- Tests, 💯

### CODE REVIEW

File: lib/resources/entityQuery.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 const Request = require('../helpers/request');
-const EntityQuery = require('../../node_modules/drupal-api/src/entityQuery');
+const EntityQuery = require('drupal-api').entityQuery;
```
Comment (mattgrill) [inline lib/resources/entityQuery.js:2]: This fixes an issue when using `waterwheel` in a server environment.

File: lib/waterwheel.js [hunk 2]
```diff
@@ -10,17 +13,11 @@ module.exports = class Waterwheel {
...
     this.credentials = credentials;
     this.api = {
-      content: new Entity(base, credentials, {get: 'node', set: 'node', create: 'entity/node', delete: 'node'}),
-      comment: new Entity(base, credentials, {get: 'comment', set: 'comment', create: 'comment', delete: 'comment'}),
-      contentType: new Entity(base, credentials, {get: '/entity/node_type', set: '/entity/node_type', create: '/entity/node_type', delete: '/entity/node_type'}),
-      file: new Entity(base, credentials, {get: 'entity/file', set: 'entity/file', create: 'entity/file', delete: 'entity/file'}),
-      menu: new Entity(base, credentials, {get: 'entity/menu', set: 'entity/menu', create: 'entity/menu', delete: 'entity/menu'}),
-      taxonomyTerm: new Entity(base, credentials, {get: 'taxonomy/term', set: 'taxonomy/term', create: 'taxonomy/term', delete: 'taxonomy/term'}),
-      taxonomyVocabulary: new Entity(base, credentials, {get: 'entity/taxonomy_vocabulary', set: 'entity/taxonomy_vocabulary', create: 'entity/taxonomy_vocabulary', delete: 'entity/taxonomy_vocabulary'}),
-      user: new Entity(base, credentials, {get: 'user', set: 'user', create: 'user', delete: 'user'}),
```
Comment (mattgrill) [inline lib/waterwheel.js:23]: 👋

File: README.md [hunk 3]
```diff
@@ -6,66 +17,207 @@ A generic JavaScript helper library to query and manipulate Drupal 8 via core RE
...
+
+```
+npm run build
+```
 
-## Usage
+## Documentation
 
-First, ensure that you have set up cross-origin resource sharing on your Drupal site to enable Waterwheel to perform necessary tasks.
+Ensure that you have set up cross-origin resource sharing on your Drupal site to enable Waterwheel to perform necessary tasks.
```
Comment (gabesullice) [inline README.md]: Can we link to some helpful resources here?

File: README.md [hunk 4]
```diff
@@ -6,66 +17,207 @@ A generic JavaScript helper library to query and manipulate Drupal 8 via core RE
...
+
+```javascript
+waterwheel.populateResources()
   .then(res => {
-    // res
+    // An array of available resources
+  });
+```
+
+Waterwheel expects the [Waterwheel-Drupal module](https://www.drupal.org/project/waterwheel) to be installed and enabled. **This must be done before any subsequent api calls**.
```
Comment (gabesullice) [inline README.md]: Should we use the waterwheel module to add a dependency on [entityqueryapi](https://www.drupal.org/project/entityqueryapi)? None of the query stuff will work without it.

File: README.md [hunk 5]
```diff
@@ -6,66 +17,207 @@ A generic JavaScript helper library to query and manipulate Drupal 8 via core RE
...
+    // err
+  });
+```
+
+`.getFieldData()` accepts no arguments. This returns the object from Waterwheel-Drupal that contains information about each field in the entity/bundle. For a list of fields, _to be used in `.setField()`_, something like `Object.keys(res.fields)` will work.
 
-Promise.all([waterwheel.api.contentType.get('article'), waterwheel.api.contentType.get('page'), waterwheel.api.content.get(1)])
+### Entity Query
+
+To take advantage of the Entity Query support, enable the [EntityQueryAPI](https://www.drupal.org/project/entityqueryapi) module. You do not need to run `.populateResources()` prior to using this functionality.
```
Comment (gabesullice) [inline README.md:226]: Disregard last comment.

File: README.md (hunk 3 above)
Comment (mattgrill) [inline README.md]: Good point. Let me look up what I did exactly and I can write something up.

File: README.md (hunk 4 above)
Comment (gabesullice) [inline README.md]: This should probably be mentioned in "Set up"

File: README.md [hunk 6]
```diff
@@ -1,3 +1,14 @@
+```
+ __      __            __                               __                     ___      
+/\ \  __/\ \          /\ \__                           /\ \                   /\_ \     
+\ \ \/\ \ \ \     __  \ \ ,_\    __   _ __   __  __  __\ \ \___      __     __\//\ \    
+ \ \ \ \ \ \ \  /'__`\ \ \ \/  /'__`\/\`'__\/\ \/\ \/\ \\ \  _ `\  /'__`\ /'__`\\ \ \   
+  \ \ \_/ \_\ \/\ \L\.\_\ \ \_/\  __/\ \ \/ \ \ \_/ \_/ \\ \ \ \ \/\  __//\  __/ \_\ \_
+   \ `\___x___/\ \__/.\_\\ \__\ \____\\ \_\  \ \___x___/' \ \_\ \_\ \____\ \____\/\____\
+    '\/__//__/  \/__/\/_/ \/__/\/____/ \/_/   \/__//__/    \/_/\/_/\/____/\/____/\/____/
+
+```
```
Comment (mattgrill) [inline README.md:10]: No project is complete without some sweet ascii art.

Comment (gabesullice) [inline README.md:10]: It would just be unprofessional _not_ to have it.

**Pull request merged**